In [4]:
!nvidia-smi

Tue May 26 08:29:41 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.82       Driver Version: 440.82       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 105...  Off  | 00000000:01:00.0  On |                  N/A |
| 30%   27C    P8    N/A /  75W |    684MiB /  4033MiB |      4%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU  

this exercise is taken from 
https://developer.download.nvidia.com/video/gputechconf/gtc/2019/presentation/s9797-dask-extensions-and-new-developments-with-rapids.pdf

In [5]:
import pandas
import cudf as pd_cudf

### Show the size of the data-set


In [16]:
!du -hs data/yellow_tripdata_2019-02.csv

620M	data/yellow_tripdata_2019-02.csv


### Read Taxi Data from February 2019
with pandas.read  

In [31]:
%time len(pandas.read_csv('data/yellow_tripdata_2019-02.csv'))

CPU times: user 14.3 s, sys: 1.94 s, total: 16.2 s
Wall time: 16.2 s


7019375

### Read Taxi Data from June 2019
with pandas.read

In [8]:
%time len(pandas.read_csv('data/yellow_tripdata_2019-06.csv'))

CPU times: user 14.6 s, sys: 1.44 s, total: 16.1 s
Wall time: 16.1 s


6941024

Typically the output looks likes this

CPU times: user 14.6 s, sys: 1.44 s, total: 16.1 s
Wall time: 16.1 s

### Read the February data with pd_cudf

In [19]:
%time len(pd_cudf.read_csv('data/yellow_tripdata_2019-02.csv'))

CPU times: user 1.6 s, sys: 242 ms, total: 1.84 s
Wall time: 1.84 s


7019375

Typically the output looks likes this

CPU times: user 1.56 s, sys: 249 ms, total: 1.81 s
Wall time: 1.81 s